lib/errorHandler.js

```diff
@@ -40,10 +40,16 @@ function handleHttpError(err, response, callback) {
   if (response.statusCode !== 200 &&
       response.statusCode !== 201 &&
       response.statusCode !== 204) {
-    var errorBody = _parseError(JSON.parse(response.body));
-    var error = createError(response.statusCode,
-                            errorBody.message,
-                            {name: _convertErrorName(errorBody.id)});
+    var error = null;
+    if(response.getHeader('content-type') == 'text/json') {
+      var errorBody = _parseError(JSON.parse(response.body));
+      error = createError(response.statusCode,
+                          errorBody.message,
+                          {name: _convertErrorName(errorBody.id)});
+    } else {
+      error = createError(response.statusCode,
+                          'InternalServerError');
```

Comment:
I would prefer still keeping the actual error message here.  Return `response.body` rather than masking it with 'InternalServerError'

Alternatively, 

```
var errorBody;
try { 
    errorBody = _parseError(JSON.parse(response.body));
 } catch (ex) {
   errorBody = { id: -1, message: response.body };
}
...